core/device_hooks/cuda_hooks.cpp

```diff
@@ -75,6 +75,10 @@ bool CudaAsyncAllocator::check_environment(int device_id,
     GKO_NOT_COMPILED(cuda);
 
 
+CudaUnifiedAllocator::CudaUnifiedAllocator(int device_id)
```

Comment:
Why did it not throw undefined reference before?

Reply:
Because we did not call that constructor anywhere, we used the one with `flags` IIRC